extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-widget.tsx

```diff
@@ -0,0 +1,161 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { Message } from '@phosphor/messaging';
+import { DisposableCollection } from '@theia/core';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import { AlertMessage } from '@theia/core/lib/browser/widgets/alert-message';
+import * as React from 'react';
+import { ChePluginMetadata, ChePluginService } from '../../common/che-protocol';
+
+@injectable()
+export class ChePluginWidget extends ReactWidget {
+
+    protected plugins: ChePluginMetadata[] = [];
+
+    protected readonly toDisposeOnFetch = new DisposableCollection();
+    protected readonly toDisposeOnSearch = new DisposableCollection();
+    protected ready = false;
+
+    protected needToBeRendered = true;
+
+    constructor(
+        @inject(ChePluginService) protected readonly chePluginService: ChePluginService
+    ) {
+        super();
+        this.id = 'che-plugins';
+        this.title.label = 'Che Plugins';
+        this.title.caption = 'Che Plugins';
+        this.title.iconClass = 'fa che-plugins-tab-icon';
+        this.title.closable = true;
+        this.addClass('theia-plugins');
+
+        this.node.tabIndex = -1;
```

Comment:
Why do we use '-1' for the tabIndex(not 0)?

Reply:
Fixed. Thanks!